File: VICSApp/CoreSystem/DHCD/FormUI/Vote/MatterVoteList.vb
Public Class MatterVoteList


    Private Sub ToolStripButton4_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ToolStripButton4.Click
        filldgv()
    End Sub

    Private Sub MatterVoteList_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Me.MdiParent = Mainform
        filldgv()
    End Sub
    Private Sub filldgv()
        Dim t As New DataTable
        Try
            Dim HolderIdentify As String
            Dim mattcode As Integer = 0
            Try
                mattcode = ToolStripTextBox1.Text
            Catch ex As Exception
                mattcode = 0
            End Try
            Try
                HolderIdentify = ToolStripTextBox2.Text
            Catch ex As Exception
                HolderIdentify = ""
            End Try

            t = Mainform.BenlyDal.MatterVotes_getlist(Mainform.workingmeeting, mattcode, HolderIdentify)

        Catch ex As Exception
            MsgBox("Lỗi :" + ex.Message)
        End Try

        Dim totalright As Integer = 0
        Dim agreecount As Integer = 0
        Dim agreeright As Integer = 0
        Dim disagreecount As Integer = 0
        Dim disagreeright As Integer = 0
        Dim noideacount As Integer = 0
        Dim noidearight As Integer = 0

        For Each dr As DataRow In t.Rows
            totalright = totalright + dr.Item("Voterights")
            If dr.Item("Agree") = True Then
                agreecount = agreecount + 1
                agreeright = agreeright + dr.Item("Voterights")
            ElseIf dr.Item("DisAgree") = True Then
                disagreecount = disagreecount + 1
                disagreeright = disagreeright + dr.Item("Voterights")
            Else
                noideacount = noideacount + 1
                noidearight = noidearight + dr.Item("Voterights")
            End If
        Next
        DataGridView1.DataSource = t
        ToolStripStatusLabel2.Text = DataGridView1.RowCount
        ToolStripStatusLabel16.Text = Mainform.addthousandseperator(totalright.ToString)
        '--------
        ToolStripStatusLabel4.Text = Mainform.addthousandseperator(agreecount.ToString) + " -- "
        ToolStripStatusLabel5.Text = Mainform.addthousandseperator(agreeright.ToString) + " -- "
        ToolStripStatusLabel8.Text = Mainform.addthousandseperator(disagreecount.ToString) + " -- "
        ToolStripStatusLabel9.Text = Mainform.addthousandseperator(disagreeright.ToString) + " -- "

        ToolStripStatusLabel12.Text = Mainform.addthousandseperator(noideacount.ToString) + " -- "
        ToolStripStatusLabel13.Text = Mainform.addthousandseperator(noidearight.ToString) + " -- "
        If totalright > 0 Then
            ToolStripStatusLabel6.Text = (Math.Round(((agreeright / totalright) * 100), 2)).ToString + "% "
            ToolStripStatusLabel10.Text = (Math.Round(((disagreeright / totalright) * 100), 2)).ToString + "% "
            ToolStripStatusLabel14.Text = (Math.Round(((noidearight / totalright) * 100), 2)).ToString + "% "
        Else
            ToolStripStatusLabel6.Text = ""
            ToolStripStatusLabel10.Text = ""
            ToolStripStatusLabel14.Text = ""
        End If

    End Sub

    Private Sub ToolStripButton1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ToolStripButton1.Click
        Dim matcode As Integer = 0
        Dim delecode As Integer = 0
        Try
            matcode = ToolStripTextBox1.Text
        Catch ex As Exception
            matcode = 0
        End Try
        Try
            delecode = ToolStripTextBox2.Text
        Catch ex As Exception
            delecode = 0
        End Try
        Dim f As New Mattervote_ins_update("Add", matcode, delecode)
        f.ShowDialog()
        filldgv()
    End Sub

    Private Sub ToolStripButton2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ToolStripButton2.Click
        Dim f As New Mattervote_ins_update("Update", DataGridView1.CurrentRow.Cells("Mattercode").Value, DataGridView1.CurrentRow.Cells("delegatecode").Value)
        f.ShowDialog()
        filldgv()
    End Sub

    Private Sub ToolStripButton3_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ToolStripButton3.Click
        If DataGridView1.SelectedRows.Count = 0 Then
            MsgBox("Bạn phải chọn ít nhất một bản ghi")
            Exit Sub
        End If
        If MsgBox("Bạn có chắc chắn XÓA Phiếu biểu quyết vấn đề :" + DataGridView1.CurrentRow.Cells("mattername").Value + " của đb " + DataGridView1.CurrentRow.Cells("Delegatename").Value, MsgBoxStyle.OkCancel + MsgBoxStyle.Critical + MsgBoxStyle.ApplicationModal + MsgBoxStyle.DefaultButton2, "XÓA PHIẾU BIỂU QUYẾT") = MsgBoxResult.Ok Then
            Try
                Mainform.BenlyDal.MatterVotes_delete(Mainform.workingmeeting, DataGridView1.CurrentRow.Cells("Mattercode").Value, DataGridView1.CurrentRow.Cells("delegatecode").Value, DataGridView1.CurrentRow.Cells("HolderCode").Value)
            Catch ex As Exception
                MsgBox("Lỗi :" + ex.Message)
            End Try
        End If
        filldgv()
    End Sub

    Private Sub MatterVoteList_KeyUp(ByVal sender As System.Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles MyBase.KeyUp
        If Not (ToolStripTextBox1.Focused Or ToolStripTextBox2.Focused) Then
            Select Case e.KeyCode
                Case Keys.A
                    ToolStripButton1_Click(sender, e)
                Case Keys.E
                    ToolStripButton2_Click(sender, e)
                Case Keys.D
                    ToolStripButton3_Click(sender, e)
                Case Keys.Escape
                    Me.Close()
            End Select
        End If

    End Sub

End Class

File: VICSApp/CoreSystem/DHCD/FormUI/Vote/MatterVoteList.cs
using System;
using System.Data;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace pmDHCD
{
    public partial class MatterVoteList
    {
        public MatterVoteList()
        {
            InitializeComponent();
            _ToolStripButton1.Name = "ToolStripButton1";
            _ToolStripButton2.Name = "ToolStripButton2";
            _ToolStripButton3.Name = "ToolStripButton3";
            _ToolStripButton4.Name = "ToolStripButton4";
        }

        private void ToolStripButton4_Click(object sender, EventArgs e)
        {
            filldgv();
        }

        private void MatterVoteList_Load(object sender, EventArgs e)
        {
            MdiParent = My.MyProject.Forms.Mainform;
            filldgv();
        }

        private void filldgv()
        {
            var t = new DataTable();
            try
            {
                string HolderIdentify;
                int mattcode = 0;
                try
                {
                    mattcode = Conversions.ToInteger(ToolStripTextBox1.Text);
                }
                catch (Exception ex)
                {
                    mattcode = 0;
                }

                try
                {
                    HolderIdentify = ToolStripTextBox2.Text;
                }
                catch (Exception ex)
                {
                    HolderIdentify = "";
                }

                t = My.MyProject.Forms.Mainform.BenlyDal.MatterVotes_getlist(My.MyProject.Forms.Mainform.workingmeeting, mattcode, HolderIdentify);
            }
            catch (Exception ex)
            {
                Interaction.MsgBox("Lỗi :" + ex.Message);
            }

            int totalright = 0;
            int agreecount = 0;
            int agreeright = 0;
            int disagreecount = 0;
            int disagreeright = 0;
            int noideacount = 0;
            int noidearight = 0;
            foreach (DataRow dr in t.Rows)
            {
                totalright = Conversions.ToInteger(Operators.AddObject(totalright, dr["Voterights"]));
                if (Conversions.ToBoolean(Operators.ConditionalCompareObjectEqual(dr["Agree"], true, false)))
                {
                    agreecount = agreecount + 1;
                    agreeright = Conversions.ToInteger(Operators.AddObject(agreeright, dr["Voterights"]));
                }
                else if (Conversions.ToBoolean(Operators.ConditionalCompareObjectEqual(dr["DisAgree"], true, false)))
                {
                    disagreecount = disagreecount + 1;
                    disagreeright = Conversions.ToInteger(Operators.AddObject(disagreeright, dr["Voterights"]));
                }
                else
                {
                    noideacount = noideacount + 1;
                    noidearight = Conversions.ToInteger(Operators.AddObject(noidearight, dr["Voterights"]));
                }
            }

            DataGridView1.DataSource = t;
            ToolStripStatusLabel2.Text = DataGridView1.RowCount.ToString();
            ToolStripStatusLabel16.Text = My.MyProject.Forms.Mainform.addthousandseperator(totalright.ToString());
            // --------
            ToolStripStatusLabel4.Text = My.MyProject.Forms.Mainform.addthousandseperator(agreecount.ToString()) + " -- ";
            ToolStripStatusLabel5.Text = My.MyProject.Forms.Mainform.addthousandseperator(agreeright.ToString()) + " -- ";
            ToolStripStatusLabel8.Text = My.MyProject.Forms.Mainform.addthousandseperator(disagreecount.ToString()) + " -- ";
            ToolStripStatusLabel9.Text = My.MyProject.Forms.Mainform.addthousandseperator(disagreeright.ToString()) + " -- ";
            ToolStripStatusLabel12.Text = My.MyProject.Forms.Mainform.addthousandseperator(noideacount.ToString()) + " -- ";
            ToolStripStatusLabel13.Text = My.MyProject.Forms.Mainform.addthousandseperator(noidearight.ToString()) + " -- ";
            if (totalright > 0)
            {
                ToolStripStatusLabel6.Text = Math.Round(agreeright / (double)totalright * 100d, 2).ToString() + "% ";
                ToolStripStatusLabel10.Text = Math.Round(disagreeright / (double)totalright * 100d, 2).ToString() + "% ";
                ToolStripStatusLabel14.Text = Math.Round(noidearight / (double)totalright * 100d, 2).ToString() + "% ";
            }
            else
            {
                ToolStripStatusLabel6.Text = "";
                ToolStripStatusLabel10.Text = "";
                ToolStripStatusLabel14.Text = "";
            }
        }

        private void ToolStripButton1_Click(object sender, EventArgs e)
        {
            int matcode = 0;
            int delecode = 0;
            try
            {
                matcode = Conversions.ToInteger(ToolStripTextBox1.Text);
            }
            catch (Exception ex)
            {
                matcode = 0;
            }

            try
            {
                delecode = Conversions.ToInteger(ToolStripTextBox2.Text);
            }
            catch (Exception ex)
            {
                delecode = 0;
            }

            var f = new Mattervote_ins_update("Add", matcode, delecode);
            f.ShowDialog();
            filldgv();
        }

        private void ToolStripButton2_Click(object sender, EventArgs e)
        {
            var f = new Mattervote_ins_update("Update", Conversions.ToInteger(DataGridView1.CurrentRow.Cells["Mattercode"].Value), Conversions.ToInteger(DataGridView1.CurrentRow.Cells["delegatecode"].Value));
            f.ShowDialog();
            filldgv();
        }

        private void ToolStripButton3_Click(object sender, EventArgs e)
        {
            if (DataGridView1.SelectedRows.Count == 0)
            {
                Interaction.MsgBox("Bạn phải chọn ít nhất một bản ghi");
                return;
            }

            if (Interaction.MsgBox(Operators.AddObject(Operators.AddObject(Operators.AddObject("Bạn có chắc chắn XÓA Phiếu biểu quyết vấn đề :", DataGridView1.CurrentRow.Cells["mattername"].Value), " của đb "), DataGridView1.CurrentRow.Cells["Delegatename"].Value), (MsgBoxStyle)((int)MsgBoxStyle.OkCancel + (int)MsgBoxStyle.Critical + (int)MsgBoxStyle.ApplicationModal + (int)MsgBoxStyle.DefaultButton2), "XÓA PHIẾU BIỂU QUYẾT") == MsgBoxResult.Ok)
            {
                try
                {
                    My.MyProject.Forms.Mainform.BenlyDal.MatterVotes_delete(My.MyProject.Forms.Mainform.workingmeeting, Conversions.ToDecimal(DataGridView1.CurrentRow.Cells["Mattercode"].Value), Conversions.ToDecimal(DataGridView1.CurrentRow.Cells["delegatecode"].Value), Conversions.ToDecimal(DataGridView1.CurrentRow.Cells["HolderCode"].Value));
                }
                catch (Exception ex)
                {
                    Interaction.MsgBox("Lỗi :" + ex.Message);
                }
            }

            filldgv();
        }

        private void MatterVoteList_KeyUp(object sender, KeyEventArgs e)
        {
            if (!(ToolStripTextBox1.Focused | ToolStripTextBox2.Focused))
            {
                switch (e.KeyCode)
                {
                    case Keys.A:
                        {
                            ToolStripButton1_Click(sender, e);
                            break;
                        }

                    case Keys.E:
                        {
                            ToolStripButton2_Click(sender, e);
                            break;
                        }

                    case Keys.D:
                        {
                            ToolStripButton3_Click(sender, e);
                            break;
                        }

                    case Keys.Escape:
                        {
                            Close();
                            break;
                        }
                }
            }
        }

        private void DataGridView1_CellContentDoubleClick(object sender, DataGridViewCellEventArgs e)
        {
            var f = new Mattervote_ins_update("Update", Conversions.ToInteger(DataGridView1.CurrentRow.Cells["Mattercode"].Value), Conversions.ToInteger(DataGridView1.CurrentRow.Cells["delegatecode"].Value));
            f.ShowDialog();
            filldgv();
        }

        private void DataGridView1_CellClick(object sender, DataGridViewCellEventArgs e)
        {
            if (DataGridView1.CurrentRow == null || e.RowIndex < 0 || e.ColumnIndex < 0) return;

            var rowIndex = e.RowIndex; //CurrentRow.Index;
            var colIndex = e.ColumnIndex;
            DataGridViewRow row = DataGridView1.Rows[rowIndex];
            var currentValue = row.Cells[colIndex]?.Value?.ToString();
            string columnName = DataGridView1.Columns[colIndex].Name.ToString();
            bool isAgree = false, isDisAgree = false, isNoidea = false;
            if (columnName == "agree" || columnName == "Disagree" || columnName == "Noidea")
            {
                bool iscurrentValue = Conversions.ToBoolean(currentValue);

                switch (columnName)
                {
                    case "agree":
                        isAgree = true;
                        isDisAgree = isNoidea = !isAgree;
                        break;
                    case "Disagree":
                        isDisAgree = true;
                        isAgree = isNoidea = !isDisAgree;
                        break;
                    case "Noidea":
                        isNoidea = true;
                        isAgree = isDisAgree = !isNoidea;
                        break;
                    default:
                        break;
                }

                try
                {
                    string meetingcode = My.MyProject.Forms.Mainform.workingmeeting;
                    decimal mattercode = Conversions.ToDecimal(row.Cells["Mattercode"].Value.ToString());
                    decimal delegatecode = Conversions.ToDecimal(row.Cells["Delegatecode"].Value.ToString());

                    My.MyProject.Forms.Mainform.BenlyDal.MatterVotes_update(meetingcode, mattercode, delegatecode, isAgree, isDisAgree, isNoidea);
                    filldgv();
                }
                catch (Exception ex)
                {
                    Interaction.MsgBox("Lỗi :" + ex.Message);
                }
            }
        }
    }
}